blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/ArtifactImpl.java

```diff
@@ -41,7 +54,24 @@ public long getSize() {
     }
 
     @Override
-    public Link getLink() {
-        return self;
+    public boolean isDownloadable() {
+        return true;
+    }
+
+    @Extension(ordinal = -1)
+    public static class FactoryImpl extends BlueArtifactFactory {
+        @Override
+        public Collection<BlueArtifact> getArtifacts(final Run<?, ?> run, final Reachable parent) {
+            // Check security for artifacts
+            if(Functions.isArtifactsPermissionEnabled() && !run.hasPermission(Run.ARTIFACTS)) {
```

Comment:
Ah, this is moved over here, but then we are leaving this check up to BlueArtifactFactory implementors. e.g. HtmlArtifact's factory is not doing this check. We should move it up to `ArtifactContainerImpl.get()` and `ArtifactContainerImpl.iterator()`